File: MyTech.Api/Requests/AddProductRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace MyTech.Api.Requests
{
    public class AddProductRequest
    {
        public string ProductText{ get; set; }
        public string SelectedProvider { get; set; }
        public double ProductPrice { get; set; }
        public int ProductStock { get; set; }
    }
}

File: my-tech/src/requests/AddProductRequest.js
import axios from 'axios';
import ApiConfig from '../ApiConfig';
import Product from '../representations/Product';

export default class AddProductRequest {
    constructor(productText, selectedProvider, productPrice, productStock) {
        this.productText = productText;
        this.selectedProvider = selectedProvider;
        this.productPrice = productPrice;
        this.productStock = productStock;
    }

    async send() {
        const response = await axios.post(ApiConfig.endpoints.products.add, {
            productText: this.productText,
            selectedProvider: this.selectedProvider,
            productPrice: this.productPrice,
            productStock: this.productStock
        });
        return Product.fromApiResponseProduct(response.data);
    }
}